components/TodoInput.tsx

```diff
@@ -0,0 +1,53 @@
+import { StyleSheet } from "react-native";
+import { useState } from "react";
+import { TextInput, View,Button,Image } from "react-native";
+
+const TodoInput  = (props) => {
+    const [enteredTodoText, setEnteredTodoText] = useState('');
+    
+
+    function enterTodoText(enteredText: string){
+        setEnteredTodoText(enteredText);
+      }
+      function onPress(){
+        props.onTodoAdded(enteredTodoText);
+        setEnteredTodoText('');
```

Comment:
Once you change your useRef technique you have to then change to direct manipulation technique from react native docs to clear the input.